sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/CountingInput.java

```diff
@@ -157,7 +172,22 @@ public UnboundedCountingInput withTimestampFn(SerializableFunction<Long, Instant
     public UnboundedCountingInput withMaxNumRecords(long maxRecords) {
       checkArgument(
           maxRecords > 0, "MaxRecords must be a positive (nonzero) value. Got %s", maxRecords);
-      return new UnboundedCountingInput(timestampFn, Optional.of(maxRecords), maxReadTime);
+      return new UnboundedCountingInput(
+          timestampFn, elementsPerPeriod, period, Optional.of(maxRecords), maxReadTime);
+    }
+
+    /**
+     * Returns an {@link UnboundedCountingInput} like this one, but with output production limited
+     * to an aggregate rate of no more than the number of elements per the period length.
+     *
+     * <p>Note that this period is taken in aggregate across all instances of the
+     * {@link PTransform}, which may cause elements to be produced in bunches.
```

Comment:
The bunches behavior is less there now, right? How about something like

> Note that when there are multiple splits, each split is given a slower rate such that the aggregate rate of production is kept the same. However, different splits are not explicitly desynchronized and so in aggregate they may not be evenly distributed across time.

Also, can we attempt to explicitly synchronize them by offseting the start time? That way if I have 1 every s split 10 ways, the start time is 1s later for the split that produces element 2, etc.

Of course we'll still have the offset in worker starting time.